File: app/src/main/java/com/volkanhotur/basemvvm/android/di/AppComponent.java
package com.volkanhotur.basemvvm.android.di;

import android.app.Application;

import com.volkanhotur.basemvvm.android.di.module.AppModule;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;
import dagger.android.support.AndroidSupportInjectionModule;

/**
 * @author volkanhotur
 */

@Singleton
@Component(modules = {AndroidSupportInjectionModule.class, AppModule.class})
public interface AppComponent extends AndroidInjector<DaggerApplication> {

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(Application application);

        AppComponent build();
    }
}

File: app/src/main/java/com/volkanhotur/basemvvm/di/AppComponent.kt
package com.volkanhotur.basemvvm.di

import android.app.Application
import com.volkanhotur.basemvvm.di.module.AppModule
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import dagger.android.support.DaggerApplication
import javax.inject.Singleton

/**
 * @author volkanhotur
 */

@Singleton
@Component(modules = [AndroidSupportInjectionModule::class, AppModule::class])
interface AppComponent : AndroidInjector<DaggerApplication> {

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }
}